### PR TITLE
Deprecate vectorized big methods in favor of compact broadcast syntax

### PR DESCRIPTION
**Pull request by Sacha0**
This PR deprecates all (?) remaining vectorized `big` methods in favor of compact broadcast syntax. Ref. #16285, #17302, and #18495.

As with #18495, to work around #18462 I made the `Broadcast` module `import` rather than `export` `broadcast`. I will correct this if / when a better solution for #18462 appears.

I imagine #17302 was a bit of a bear to review. So for devectorization of manually vectorized functions, I'm shooting for small, atomic PRs with the hope that that will ease review. If instead I should bundle them, please let me know! Best!

### CODE REVIEW

File: base/float.jl [hunk 1]
```diff
@@ -554,5 +554,15 @@ for fn in (:float,:big)
...
+# big, broadcast over arrays
+# TODO: do the definitions below primarily pertaining to integers belong in float.jl?
+function big end # no prior definitions of big in sysimg.jl, necessitating this
+broadcast{T<:Integer,N}(::typeof(big), x::AbstractArray{T,N}) = convert(AbstractArray{BigInt,N}, x)
+broadcast{T<:AbstractFloat,N}(::typeof(big), x::AbstractArray{T,N}) = convert(AbstractArray{BigFloat,N}, x)
+broadcast(::typeof(big), r::UnitRange) = big(r.start):big(last(r))
+broadcast(::typeof(big), r::StepRange) = big(r.start):big(r.step):big(last(r))
+broadcast(::typeof(big), r::FloatRange) = FloatRange(big(r.start), big(r.step), r.len, big(r.divisor))
+function broadcast(::typeof(big), r::LinSpace)
+    big(r.len) == r.len || error(string(r, ": too long for ", big))
```
Comment (ararslan) [inline base/float.jl]: Might be better as an `ArgumentError`?

File: base/float.jl [hunk 2]
```diff
@@ -541,7 +541,7 @@ function float{T}(A::AbstractArray{T})
     convert(AbstractArray{typeof(float(zero(T)))}, A)
 end
 
-for fn in (:float,:big)
+for fn in (:float,)
```
Comment (ararslan) [inline base/float.jl]: You could probably just deloopify this since it's iterating over a single value, or so I would assume.

**Comment by ararslan** [comment]
You always do such great work, Sacha!

File: base/float.jl (hunk 2 above)
Comment (Sacha0) [inline base/float.jl]: Absolutely. I anticipate removing this block altogether when rebasing with #18495 in (or removing the corresponding block from #18495 when rebasing #18495 with this PR in, whichever occurs).

File: base/float.jl (hunk 1 above)
Comment (Sacha0) [inline base/float.jl]: Good call. Changing here and in #18495. Thanks!

**Comment by Sacha0** [comment]
Thanks for the kind words @ararslan! :blush:

File: base/complex.jl [hunk 3]
```diff
@@ -828,8 +828,8 @@ function complex{T}(A::AbstractArray{T})
     convert(AbstractArray{typeof(complex(zero(T)))}, A)
 end
 
-big{T<:Integer,N}(A::AbstractArray{Complex{T},N}) = convert(AbstractArray{Complex{BigInt},N}, A)
-big{T<:AbstractFloat,N}(A::AbstractArray{Complex{T},N}) = convert(AbstractArray{Complex{BigFloat},N}, A)
+broadcast{T<:Integer,N}(::typeof(big), A::AbstractArray{Complex{T},N}) = convert(AbstractArray{Complex{BigInt},N}, A)
```
Comment (TotalVerb) [inline base/complex.jl]: We could possibly document that `broadcast` can alias, like we do for `convert`. https://github.com/JuliaLang/julia/pull/17519

Comment (Sacha0) [inline base/complex.jl]: Ref. ongoing discussion in #18495, and particularly https://github.com/JuliaLang/julia/pull/18495#issuecomment-247491332. Agreed, that is :).

**Comment by Sacha0** [comment]
Rebased. Best!

**Comment by Sacha0** [comment]
Subsumed by #19791.

**Pull request closed without merging**